packages/core/useScriptTag/index.ts

```diff
@@ -0,0 +1,124 @@
+import { isString, MaybeRef, tryOnMounted, tryOnUnmounted } from '@vueuse/shared'
+import { ref } from 'vue-demi'
+import { ConfigurableDocument, defaultDocument } from '../_configurable'
+
+export interface UseScriptTagOptions extends ConfigurableDocument {
+  immediate?: boolean
+}
+
+/**
+ * Async script tag loading.
+ *
+ * @see   {@link https://vueuse.js.org/useScriptTag}
+ * @param src
+ */
+export function useScriptTag(
+  src: MaybeRef<string>,
+  onLoaded: Function = () => {},
+  {
+    immediate = true,
+    document = defaultDocument,
+  }: UseScriptTagOptions = {},
+) {
+  const scriptTag = ref<HTMLScriptElement>()
+
+  /**
+   * Load the script specified via `src`.
+   *
+   * @param waitForScriptLoad Whether if the Promise should resolve once the "load" event is emitted by the <script> attribute, or right after appending it to the DOM.
+   * @returns Promise<HTMLScriptElement>
+   */
+  const loadScript = (waitForScriptLoad = true): Promise<HTMLScriptElement> =>
+    new Promise((resolve, reject) => {
+      if (!document) {
+        reject(new Error('No document found!'))
+        return
+      }
+
+      if (!src) {
+        reject(new Error('No src found!'))
+        return
+      }
+
+      let shouldAppend = false
+
+      let el: HTMLScriptElement = <HTMLScriptElement>(
+        document.querySelector(`script[src="${src}"]`)
+      )
+
+      if (!el) {
+        el = document.createElement('script')
+        el.type = 'text/javascript'
+        el.async = true
```

Comment:
Should we make this configurable? Also for `crossorigin` `defer` `referrerpolicy` etc.

https://www.w3schools.com/tags/tag_script.asp

Reply:
```
  async?: boolean
  type?: string
  crossOrigin?: 'anonymous' | 'use-credentials'
  referrerPolicy?: 'no-referrer' | 'no-referrer-when-downgrade' | 'origin' | 'origin-when-cross-origin' | 'same-origin' | 'strict-origin' | 'strict-origin-when-cross-origin' | 'unsafe-url'
  noModule?: boolean
  defer?: boolean
  ```
  
  I've added all these parameters, I think you're right we should make everything configurable.

Reply:
Great! Thanks